src/build_options.rs

```diff
@@ -699,31 +710,77 @@ pub fn find_interpreter(
                             }
                         })
                         .context("unsupported Python interpreter")?;
-                    interpreter = vec![PythonInterpreter {
-                        major,
-                        minor,
-                        abiflags,
+                    interpreters.push(PythonInterpreter {
+                        config: InterpreterConfig {
+                            major,
+                            minor,
+                            interpreter_kind,
+                            abiflags,
+                            ext_suffix: ext_suffix.to_string(),
+                            abi_tag,
+                            calcsize_pointer: None,
+                        },
                         target: target.clone(),
                         executable: PathBuf::new(),
-                        ext_suffix: ext_suffix.to_string(),
-                        interpreter_kind,
-                        abi_tag,
                         platform: None,
                         runnable: false,
-                    }];
+                    });
+                } else {
+                    if interpreter.is_empty() {
+                        bail!("Couldn't find any python interpreters. Please specify at least one with -i");
+                    }
+                    for interp in interpreter {
+                        let python = interp
+                            .file_name()
+                            .context("Invalid python interpreter")?
+                            .to_string_lossy();
+                        let (python_impl, python_ver) =
+                            if let Some(ver) = python.strip_prefix("pypy") {
+                                (InterpreterKind::PyPy, ver)
+                            } else if let Some(ver) = python.strip_prefix("python") {
+                                (InterpreterKind::CPython, ver)
+                            } else {
+                                bail!("Unsupported Python interpreter: {}", python);
+                            };
+                        let mut ver_parts = python_ver.split('.');
```

Comment:
You might want to use `split_once` (if msrv 1.52 or above), or check `ver_parts.next()` is none after parsing minor version, so that strings like 3.7.foobar.baz aren't accepted